### ISSUE TITLE
Using Alva 640 Comfort with NVDA

### ISSUE THREAD

**Issue by Pickl** [nvaccess/nvda#9568]
Hi,

I am new to this site so unsure if I am posting in the correct place. I am hoping someone might be able to give me some pointers.

I am unable to get the Alva 640 Comfort and NVDA to communicate with each other. Any advice would be greatly received!

Currently I have ensured I have the most recent version of NVDA installed and have added some extra drivers here: C:\program files (x86)\NVDA

Thanks,

Laura

**Comment by leonardder** [comment]
Hey Laura,

Thanks for your report. As far as I know, the ALVA Comfort should work properly with the most recent version of NVDA, without any drivers. Therefore, please test with a virgin version of NVDA if you want to make sure your environment is correct.

When your ALVA is connected to your machine, could you make sure that the Optelec ALVA BC6 series display driver is listed in the list of braille displays available> The quickest way to see that list, is by pressing NVDA+ctrl+a, where the NVDA key is either the caps lock or insert key, depending on your setup.

**Comment by Pickl** [comment]
Hi!

Thank you so much for your quick response! Having just reinstalled NVDA for the 3rd time today and having your confirmation that it should work I have now discovered that it was the cable I was using that is faulty and I am now connected via bluetooth. :)

Thanks again!

**Comment by leonardder** [comment]
Thanks again.

While at it, would you be able to confirm that the Braille keyboard is 
working as expected as well?

**Comment by Pickl** [comment]
This is my first time using the ALVA so I am just trying to identify what keys do what with NVDA at the moment. 
For example, looking at the NVDA user guide I expected the NVDA Menu to open when I pressed space with dot 1 and dot 3 however this doesn't happen.
I am unsure if this is because I have installed using the Windows bluetooth windows pairing control panel applet rather than the ALVA Bluetooth Utility. I do not have another cable to test with currently.

**Comment by Pickl** [comment]
Just to update...

I am unable to change from uncontracted to contracted braille i'm trying to use UEB.

These functions work:

Enter = enter key
Windows = windows key
Backspace = Space + dot 1 + 2
Escape = Space + dot 1 + 5
Delete = Space + dot 4 + 5
Home = Space + dot 1 + 2 + 3
End = Space + dot 4 + 5 + 6
Tab = Space + dot 5 + 6

If anyone can shine any light on toggling between contracted and uncontracted UEB that would be greatly appreciated!

**Comment by leonardder** [comment]
@Pickl: You should be able to switch from contracted to uncontracted UEB by choosing Unified English Braille Code grade 2 or grade 1 respectively. This can be done using the braille category in NVDA's settings dialog.

With regarding to opening the NVDA menu, this can be done using sp1+sp3, the first and third smart pad keys.

As your display is working as expected, I"m closing this issue as works for now. Please direct further questions to [the NVDA users mailing list](https://nvda.groups.io/g/nvda).
